core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/docker/DockerClient.scala

```diff
@@ -98,6 +99,20 @@ class DockerClient(dockerHost: Option[String] = None,
     Seq(dockerBin) ++ host
   }
 
+  // Determine the version of the docker client cli.
+  // Failure to determine the client version is a fatal error, because it indicates a high probability
+  // that subsequent docker cli commands will also fail.
+  protected def getClientVersion(): String = {
+    val vf = executeProcess(dockerCmd ++ Seq("version", "--format", "{{.Client.Version}}"), config.timeouts.version)
+      .andThen {
+        case Success(version) => log.info(this, s"Detected docker client version $version")
+        case Failure(e) =>
+          log.error(this, s"Failed to determine docker client version: ${e.getClass} - ${e.getMessage}")
+      }
+    Await.result(vf, Duration.Inf) // The timeout in executeProcess ensures progress; don't create racy double timeout in Await
```

Comment:
I suggest to wait for at most 2 * `config.timeouts.version`. This should never be a race condition and still ensure we have an upper bound.

Reply:
done